pkg/maintainer/spv/spv.go

```diff
@@ -95,26 +112,58 @@ func (sm *spvMaintainer) maintainSpv(ctx context.Context) error {
 	}
 }
 
-func (sm *spvMaintainer) proveDepositSweepTransactions() error {
-	depositSweepTransactions, err := sm.getUnprovenDepositSweepTransactions()
+// unprovenTransactionsGetter is a type representing a function that is
+// used to get unproven Bitcoin transactions.
+type unprovenTransactionsGetter func(
+	historyDepth uint64,
+	transactionLimit int,
+	btcChain bitcoin.Chain,
+	spvChain Chain,
+) (
+	[]*bitcoin.Transaction,
+	error,
+)
+
+// transactionProofSubmitter is a type representing a function that is used
+// to submit the constructed SPV proof to the host chain.
+type transactionProofSubmitter func(
+	transactionHash bitcoin.Hash,
+	requiredConfirmations uint,
+	btcChain bitcoin.Chain,
+	spvChain Chain,
+) error
+
+// proveTransactions gets unproven Bitcoin transactions using the provided
+// unprovenTransactionsGetter, build the SPV proofs, and submits them using
+// the provided transactionProofSubmitter.
+func (sm *spvMaintainer) proveTransactions(
+	unprovenTransactionsGetter unprovenTransactionsGetter,
+	transactionProofSubmitter transactionProofSubmitter,
+) error {
+	transactions, err := unprovenTransactionsGetter(
+		sm.config.HistoryDepth,
+		sm.config.TransactionLimit,
+		sm.btcChain,
+		sm.spvChain,
+	)
 	if err != nil {
 		return fmt.Errorf(
-			"failed to get unproven deposit sweep transactions: [%v]",
+			"failed to get unproven transactions: [%v]",
 			err,
```

Comment:
The line is probably shorter than 80 chars now.
```suggestion
			return fmt.Errorf("failed to get unproven transactions: [%v]", err)
```

Reply:
Addressed in 1953e4a0